kotlinpoet/src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -295,14 +295,21 @@ class TypeSpec private constructor(
     for (property in propertySpecs) {
       val parameter = primaryConstructor.parameter(property.name) ?: continue
       if (parameter.type != property.type) continue
-      if (CodeBlock.of("%N", parameter) != CodeBlock.of("%N", property.initializer.toString()))
+      if (!isPropertyInitializerConstructorParameter(property, parameter))
         continue
 
       result[property.name] = property.fromPrimaryConstructorParameter(parameter)
     }
     return result
   }
 
+  /**
+   * Returns `true if the property can be declared inline as a constructor parameter
+   */
+  private fun isPropertyInitializerConstructorParameter(property: PropertySpec, parameter: ParameterSpec): Boolean {
```

Comment:
nit: Feels like these lines are too long, I'm surprised that Spotless didn't catch it.

Reply:
I guess it's just short enough. Should be better now though